README.md

```diff
@@ -1,16 +1,39 @@
-## Threadx Demo
-- Using stmcubeide version: 1.15.1 and an STM32f446RE
-- RTOS code is in Threadx-Demo/Core/Src/app_threadx.c
-- Using 3 threads with normal priority
-- Thread0 
-	- loads a message into a queue
-	- When the message is loaded a thread 0 specific count increments by 1
-	- Thread0 sum is the sum of all the random numbers it sends
-	- Thread0 prints every global piece of data
-- Thread1 and thread2 
-	- Each take the message with the random number they recieved and print it over uart
-	- Thread1 sum is the sum of the numbers it recieves
-	- Shared count is a count of the total messages sent
-	- Shared sum is a sum of all the random numbers sent
-	- When the message is recieved the threads increment a gloabl variable by 1
-	- Thread1 and 2 print their resepective counts and messages recieved over uart
+# ThreadX Demo
+- Using STM32CubeIDE version 1.15.1 and an STM32F446RE
+- RTOS code is in `ThreadX-Demo/Core/Src/app_ThreadX.c`
+- Using 3 threads with normal priority:
+
+The point of this demo is to have an example of how we can create and use threads in ThreadX.
+
+## Thread 0:
+ 1. Generates a random number and, using tx_queue_send, adds it to a queue.
+ 2. Waits to be able to take a counting semaphore(When the code is first flashed the semaphore starts in the released state so thread 0 should always get it first).
+ 3. If the random number is successfully added to the queue a thread 0 specific count increments by 1 and the random number is added to a thread 0 sum.
+ 4. Releases a counting semaphore.
+ 5. tx_thread_sleep(100) makes the thread sleep for 100 ticks or 1 second.
+ 6. Every time the count is incremented 10 times it calculates the average of all thread data and then prints the average along with all other stats in the RTOS.
+ 
+ ## Thread 1/2:
+ 1. After thread0 releases the semaphore for the first time thread 1 and thread 2 are created.
+ 2. Thread 1 waits to receive a semaphore signal from thread 2 to ensure thread 1 does not start the process before thread 2 is created. Otherwise, thread 1 would go through its process twice before thread 2 goes once.
+ 3. Inside thread 1 tx_queue_receive waits forever until it has received a message from the queue.
+ 4. Wait until the semaphore is released.
+ 5. If tx_queue_receive has received the random number, thread 1 increments a global and thread 1 count and adds the random number to a global and thread 1 sum.
+ 6. Print out the random number it received, a thread 1 count, and a thread 1 sum over UART.
+ 7. After thread 1 is done printing, it releases the semaphore 
+ 8. Thread 0 goes through the message-sending process again.
+ 9. After thread0 releases the semaphore a second time thread 2 goes through steps 2-6 with its variables.
```

Comment:
Did you make it so thread 2 specifically waits for thread 1 before running or is this just the behavior of the RTOS? Do threads 1 and 2 do the same thing?